packages/web/src/common/store/notifications/refreshNotificationsSaga.ts

```diff
@@ -0,0 +1,47 @@
+import { getErrorMessage, notificationsActions } from '@audius/common'
+import moment from 'moment'
+import { call, put, takeLatest } from 'typed-redux-saga'
+
+import { NOTIFICATION_LIMIT_DEFAULT } from './constants'
+import { fetchNotifications } from './fetchNotifications'
+import { parseAndProcessNotifications } from './parseAndProcessNotifications'
+
+const { REFRESH_NOTIFICATIONS } = notificationsActions
+
+const {
+  fetchNotificationsRequested,
+  setNotifications,
+  fetchNotificationsFailed
+} = notificationsActions
+
+function* refreshNotifications() {
+  yield* put(fetchNotificationsRequested())
+  const limit = NOTIFICATION_LIMIT_DEFAULT
+  const timeOffset = moment().toISOString()
+  const notificationsResponse = yield* call(fetchNotifications, {
+    limit,
+    timeOffset
+  })
+
+  if ('error' in notificationsResponse) {
+    yield* put(
+      fetchNotificationsFailed(getErrorMessage(notificationsResponse.error))
+    )
+    return
+  }
+
+  const { notifications: notificationItems, totalUnviewed } =
+    notificationsResponse
+
+  const notifications = yield* call(
+    parseAndProcessNotifications,
+    notificationItems
+  )
+
+  const hasMore = notifications.length >= limit
+  yield* put(setNotifications(notifications, totalUnviewed, hasMore))
+}
+
+export function* watchRefreshNotifications() {
```

Comment:
is this registered anywhere?

Reply:
oh great catch, added on a later pr, but should update here too

Reply:
yeah its in the notification/sagas.ts file, but that is not expanded cause so many lines removed